### PR TITLE
Refactored Codeception tests

### PR DESCRIPTION
**Pull request by DavertMik**
* Codeception tests are moved to `tests` dir
* `codeception/base` added as dev dependency
* removed redundant fixtures config
* removed `codeception/specify` usage
* removed `yiisoft/yii-codeception` from dependencies
* acceptance tests configured to work with WebDriver and disabled by default
* one config for all test suites is prepared and stored in `config/test.php`
* `tests/_bootstrap.php` simplified
* unit, functional and acceptance tests refactored following modern best practices. Thanks to @kosm-o and @MaxHero
* documentation in `tests/Readme.md` moved to `Readme.md` and updated

This PR requires Codeception itself to be updated before it could be merged. 
So not to merge before Codeception 2.1.3 released

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,3 +100,83 @@ return [
...
+
+Tests are located in `tests` directory, developed with [Codeception PHP Testing Framework](http://codeception.com/).
+By default there are 3 test suites: `unit`, `functional` and `acceptance`. Tests can be executed by running
+
+   ```
+   composer exec codecept run
+   ``` 
+
+This will execute unit and functional tests. Unit tests are testing the system components, while functional tests are for testing user interaction.
+Acceptance tests are disabled by default as they require additional setup as they perform testing in real browser.
```
Comment (samdark) [inline README.md]: `as they` used two times. Better to change 2nd occurrence to `since`.

File: composer.json [hunk 2]
```diff
@@ -12,15 +12,16 @@
         "irc": "irc://irc.freenode.net/yii",
         "source": "https://github.com/yiisoft/yii2"
     },
-    "minimum-stability": "dev",
+    "minimum-stability": "stable",
```
Comment (samdark) [inline composer.json]: Should be `dev`. Automatically changed to `stable` when release is tagged.

File: composer.json [hunk 3]
```diff
@@ -12,15 +12,16 @@
...
     "require": {
         "php": ">=5.4.0",
         "yiisoft/yii2": ">=2.0.5",
         "yiisoft/yii2-bootstrap": "*",
         "yiisoft/yii2-swiftmailer": "*"
     },
     "require-dev": {
-        "yiisoft/yii2-codeception": "*",
+        "codeception/base": "^2.2.3",
+        "codeception/verify": "*",
```
Comment (samdark) [inline composer.json]: Isn't it a good idea to specify version constraint for it as well?

File: config/test.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    'id' => 'basic-tests',
+    'basePath' => dirname(__DIR__),    
+    'language' => 'en-US',
+    'components' => [
+        'db' => $dbParams,
+        'mailer' => [
+            'useFileTransport' => true,
+        ],
+        'urlManager' => [
+            'showScriptName' => true,
```
Comment (samdark) [inline config/test.php]: Is it required?

Comment (samdark) [inline config/test.php]: Also, it's default value...

File: tests/functional/ContactFormCest.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        $I->see('Email cannot be blank');
+        $I->see('Subject cannot be blank');
+        $I->see('Body cannot be blank');
+        $I->see('The verification code is incorrect');
+    }
+
+    public function submitFormWithIncorrectEmail(\FunctionalTester $I)
+    {
+        $I->submitForm('#contact-form', [
+            'ContactForm[name]'			=>	'tester',
```
Comment (samdark) [inline tests/functional/ContactFormCest.php]: Weird formatting...

File: tests/functional/ContactFormCest.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        $I->dontSee('Name cannot be blank', '.help-inline');
+        $I->see('Email is not a valid email address.');
+        $I->dontSee('Subject cannot be blank', '.help-inline');
+        $I->dontSee('Body cannot be blank', '.help-inline');
+        $I->dontSee('The verification code is incorrect', '.help-inline');        
+    }
+
+    public function submitFormSuccessfully(\FunctionalTester $I)
+    {
+        $I->submitForm('#contact-form', [
```
Comment (samdark) [inline tests/functional/ContactFormCest.php]: Weird formatting...

**Comment by samdark** [comment]
Perfect. Thank you @DavertMik, @kosm-o and @MaxHero! I'll merge it after Codeception release.

**Comment by DavertMik** [comment]
It is green!

File: composer.json (hunk 3 above)
Comment (DavertMik) [inline composer.json]: yes, it can be. Current one is 0.3.1... I think ~0.3.x should be well

**Comment by samdark** [comment]
Excellent! I'll review it one more time and them merge it.

**Comment by samdark** [comment]
Merged with minor adjustments https://github.com/yiisoft/yii2-app-basic/commit/d7df546502eb04141b1d32d1e4cb705766da296d

**Comment by samdark** [comment]
Thank you very much!

**Comment by michaelarnauts** [comment]
The instructions for remote code coverage is gone now. Is this not supported anymore?

**Comment by samdark** [comment]
@DavertMik ?

File: tests/acceptance/HomeCest.php [hunk 7]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+class HomeCest
+{
+    public function ensureThatHomePageWorks(AcceptanceTester $I)
+    {
+        $I->amOnPage('index.php?r=site%2Findex');
```
Comment (ElisDN) [inline tests/acceptance/HomeCest.php]: Fix to `Url::toRoute('/site/index')` please.

Comment (DavertMik) [inline tests/acceptance/HomeCest.php]: Yes, this should be fixed. Thanks

**Comment by DavertMik** [comment]
> The instructions for remote code coverage is gone now. Is this not supported anymore?

They are not required, actually. Most of functionality should be tested with unit and functional tests, and not the acceptance. Yii performs really great in functional testing so you can collect all the coverage from it. Remote code coverage is a complicated setup and should not be included in `basic`. However, you can still use it, the [guide is still available](http://codeception.com/docs/11-Codecoverage) at Codeception site.

File: tests/acceptance/HomeCest.php (hunk 7 above)
Comment (samdark) [inline tests/acceptance/HomeCest.php]: https://github.com/yiisoft/yii2-app-basic/commit/a322cdb0a568ab66100abc63dc9d57fa5c17402d

**Pull request merged**